python/turbine_models/custom_models/stateless_llama.py

```diff
@@ -277,109 +278,28 @@ def forward(token0: torch.Tensor, *state0_flat):
-    device_inputs = [ireert.asdevicearray(config.device, example_input_id)]
 
-    ModuleCompiled = ctx.modules.state_update
-    results = ModuleCompiled["run_initialize"](*device_inputs)
+# if you're looking for run_vmfb_comparison, it's now in python/turbine_models/tests/vmfb_comparison.py
 
-    def format_out(results):
-        return torch.tensor(results.to_host()[0][0])
+if __name__ == "__main__":
+    args = parser.parse_args()
 
-    model = AutoModelForCausalLM.from_pretrained(
+    mod_str, _ = export_transformer_model(
         args.hf_model_name,
-        torch_dtype=torch.float,
-        use_auth_token=args.hf_auth_token,
+        args.hf_auth_token,
+        args.compile_to,
+        args.external_weights,
+        args.external_weight_file,
+        args.quantization,
+        args.precision,
+        args.device,
+        args.iree_target_triple,
+        args.vulkan_max_allocation,
     )
-
-    def get_token_from_logits(logits):
-        return torch.argmax(logits[:, -1, :], dim=1)
-
-    base_model_results = model.forward(example_input_id)
-    base_model_token = get_token_from_logits(base_model_results.logits)
-    bm_pkv = base_model_results.past_key_values
-    turbine_results = []
-    torch_results = []
-    turbine_results.append(format_out(results))
-    torch_results.append(int(base_model_token))
-    while base_model_token != 2:
-        results = ModuleCompiled["run_forward"](results)
-        base_model_results = model.forward(
-            torch.unsqueeze(base_model_token, 0), past_key_values=bm_pkv
-        )
-        base_model_token = get_token_from_logits(base_model_results.logits)
-
-        bm_pkv = base_model_results.past_key_values
-        # uncomment to see tokens as they are emittd
-        # print(f"pytorch: {tokenizer.decode(base_model_token)}")
-        # print(f"turbine: {tokenizer.decode(format_out(results))}")
-        turbine_results.append(format_out(results))
-        torch_results.append(int(base_model_token[0]))
-
-    print("turbine output: ")
-    print(tokenizer.decode(turbine_results))
-    print("\ntorch output: ")
-    print(tokenizer.decode(torch_results))
-
-
-if __name__ == "__main__":
-    args = parser.parse_args()
-    if args.run_vmfb:
-        run_vmfb_comparison(args)
-    else:
-        mod_str, _ = export_transformer_model(
-            args.hf_model_name,
-            args.hf_auth_token,
-            args.compile_to,
-            args.external_weights,
-            args.external_weight_file,
-            args.quantization,
-            args.precision,
-            args.device,
-            args.iree_target_triple,
-            args.vulkan_max_allocation,
-        )
-        safe_name = args.hf_model_name.split("/")[-1].strip()
-        safe_name = re.sub("-", "_", safe_name)
-        with open(f"{safe_name}.mlir", "w+") as f:
-            f.write(mod_str)
-        print("Saved to ", safe_name + ".mlir")
+    safe_name = args.hf_model_name.split("/")[-1].strip()
```

Comment:
This safe name is too generic. Can you add more detail like device name, precision, and quantization to it? Maybe even a date of when it was created?